pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -46,8 +46,11 @@ const (
 	ProviderName = "baremetal"
 	// HostAnnotation is the key for an annotation that should go on a Machine to
 	// reference what BareMetalHost it corresponds to.
-	HostAnnotation = "metal3.io/BareMetalHost"
-	requeueAfter   = time.Second * 30
+	HostAnnotation                  = "metal3.io/BareMetalHost"
+	requeueAfter                    = time.Second * 30
+	externalRemediationAnnotation   = "host.metal3.io/external-remediation"
+	remediationInProgressAnnotation = "remediation.metal3.io/remediation-in-progress"
```

Comment:
Following up on my comments below, I'll suggest name changes for consideration that I think will make this more easily understandable both by people reading the code and people interacting with the kube resources:

* rename `reboot.metal3.io/machine-remediation` to `reboot.metal3.io/requested-power-off`
* rename `remediation.metal3.io/remediation-in-progress` to `reboot.metal3.io/power-transitioned-off`

Maybe those aren't exactly the best, so feel free to suggest otherwise, but I think those names would be much more relevant to what the annotations actually signify.

Reply:
> rename `reboot.metal3.io/machine-remediation` to `reboot.metal3.io/requested-power-off`

I want the annotation to contain something that is specific for this use-case. Other controllers might add their own reboot annotation to a host, and I don't won't that two controllers will fight on the same annotation. `reboot.metal3.io/requested-power-off` sounds too generic to me. What about `reboot.metal3.io/capbm-requested-power-off`? or `reboot.metal3.io/machine-remediation-requested-power-off` ?

> rename `remediation.metal3.io/remediation-in-progress` to `reboot.metal3.io/power-transitioned-off`

I find `reboot.metal3.io/power-transitioned-off` a bit confusing. all `reboot.metal3.io` annotations signal a power off and should be placed on the host rather a machine. 
Can you explain why `remediation-in-progress` is not clear? Maybe it will help me to come up with a better name.
From my point of view, it signals that this machine is under remediation process.

Reply:
> What about reboot.metal3.io/capbm-requested-power-off? or reboot.metal3.io/machine-remediation-requested-power-off ?
I wouldn't object to either. But since these are all namespaced to metal3.io, and we don't have very many controllers, I think it's perfectly safe to just define one annotation-based reboot API and expect everyone to use it the same way.

> Can you explain why remediation-in-progress is not clear? 

As I understand the code, this annotation is not added until after 1) power off has been requested, 2) power has been observed to transition from on to off.

I'm not sure why those two steps aren't considered part of the remediation process. As a layperson, I would assume that remediation is "in progress" as soon as power is requested to be off.

So rather than try to deal with the philosophical question of what "in progress" means ;), I think it's much more clear to name it what it really is. You put that annotation on only after observing power transition from on to off.

Reply:
Also, the logic depends on this annotation at some point. Trying to understand the code and the workflow, it was hard for me to figure out why intermediate logic cares about whether things are "in progress". It makes much more sense to understand that the code is waiting for something to happen (for power to transition to off) before proceeding to the next step.

Reply:
No, the API for reboot is defined in BMO. CAPBM is only one user of this API. Others might use it as well.
We already saw some interest from the UI team to include a "reboot" button in the UI which will use this BMO reboot API. They will have to create their own annotation with some unique name.

Reply:
Quoting [reboot interface design doc](https://github.com/metal3-io/metal3-docs/blob/master/design/reboot-interface.md):
"Callers indicate to the controller that they are ready to continue by removing the annotation with their **unique** {key} suffix."

Reply:
> No, the API for reboot is defined in BMO. CAPBM is only one user of this API. Others might use it as well.
> We already saw some interest from the UI team to include a "reboot" button in the UI which will use this BMO reboot API. They will have to create their own annotation with some unique name.

The host API is for this controller to use. Everything else should be operating on Nodes or Machines so that it works for all platforms.

Reply:
> The host API is for this controller to use. Everything else should be operating on Nodes or Machines so that it works for all platforms.

I can imagine platform-specific features. Anyway, how this settle with the reboot design doc which mention that "multiple clients" are a valid use-case?

Reply:
I never agreed with that requirement, but it seemed necessary to move ahead. *shrug*

Reply:
:)
Okay, I think we all have understanding of the situation and we need to move ahead with this one as well.

We want to replace `reboot.metal3.io/machine-remediation` with some other name.
@dhellmann, What about reboot.metal3.io/capbm-requested-power-off? or reboot.metal3.io/machine-remediation-requested-power-off ?

It seems the Michael is not objecting.
Let me know what's your preference, or if you want the non-unique name (i.e. `reboot.metal3.io/requested-power-off`).

Reply:
@mhrivnak, as for remediation-in-progress, thanks for the explanation. I understand the confusion it creates, and I'll try to think on a better approach

Reply:
@mhrivnak, changed `remediation-in-progress` to `powered-off-for-remediation`

Reply:
Changed reboot annotation to `reboot.metal3.io/capbm-requested-power-off`. I believe this resolve this discussion